entrypoint.sh

```diff
@@ -30,13 +30,26 @@ export SCORECARD_POLICY_FILE="/policy.yml" # Copied at docker image creation.
 export SCORECARD_RESULTS_FILE="$INPUT_RESULTS_FILE"
 export SCORECARD_RESULTS_FORMAT="$INPUT_RESULTS_FORMAT"
 export SCORECARD_PUBLISH_RESULTS="$INPUT_PUBLISH_RESULTS"
-# https://docs.github.com/en/actions/learn-github-actions/environment-variables
-export SCORECARD_PRIVATE_REPOSITORY="$(jq '.repository.private' $GITHUB_EVENT_PATH)"
-export SCORECARD_DEFAULT_BRANCH="refs/heads/$(jq -r '.repository.default_branch' $GITHUB_EVENT_PATH)"
+export SCORECARD_REPOSITORY="$(jq -r '.repository.full_name' $GITHUB_EVENT_PATH)"
 export SCORECARD_BIN="/scorecard"
 export ENABLED_CHECKS=
 
-# WARNING: boolean inputs are strings https://github.com/actions/runner/issues/1483.
+## ============================== WARNING ======================================
+# https://docs.github.com/en/actions/learn-github-actions/environment-variables
+# export SCORECARD_PRIVATE_REPOSITORY="$(jq '.repository.private' $GITHUB_EVENT_PATH)"
+# export SCORECARD_DEFAULT_BRANCH="refs/heads/$(jq -r '.repository.default_branch' $GITHUB_EVENT_PATH)"
+#
+# The $GITHUB_EVENT_PATH file produces:
+# private: null
+# default_branch: null
+#
+# for trigger event `schedule`. This is a bug.
+# So instead we use the REST API to retrieve the data.
+#
+# Boolean inputs are strings https://github.com/actions/runner/issues/1483.
+# ===============================================================================
+export SCORECARD_PRIVATE_REPOSITORY="$(curl -s -H \"Authorization: Bearer $GITHUB_AUTH_TOKEN\" https://api.github.com/repos/$SCORECARD_REPOSITORY | jq -r '.private')"
+export SCORECARD_DEFAULT_BRANCH="refs/heads/$(curl -s -H \"Authorization: Bearer $GITHUB_AUTH_TOKEN\" https://api.github.com/repos/$SCORECARD_REPOSITORY | jq -r '.default_branch')"
```

Comment:
Might I recommend something like below instead:
```suggestion
curl -s -H "Authorization: Bearer $GITHUB_AUTH_TOKEN" https://api.github.com/repos/$SCORECARD_REPOSITORY > repo_info.json
export PR="$(cat repo_info.json | jq -r '.private')"
export DB="refs/heads/$(cat repo_info.json | jq -r '.default_branch')"
rm repo_info.json
```

This reduces the API requests by 1 and just extracts the needed information from a file. I've run into API rate-limiting issues calling the GH API from within GitHub Actions (I am an GH action author, myself), so I figured this might be a nice improvement/optimization for this action, as well.

Reply:
Good call! Will do.